modules/react-native-wallet/android/src/main/java/com/wallet/WalletModule.kt

```diff
@@ -0,0 +1,304 @@
+    if (!ensureTapAndPayClientInitialized()) {
+      promise.reject("Initialization error", "TapAndPay client initialization failed")
+      return
+    }
+
+    tapAndPayClient!!.listTokens()
+      .addOnCompleteListener { task ->
+        if (!task.isSuccessful || task.result == null) {
+          promise.reject("error", "no tokens available")
+          return@addOnCompleteListener
+        }
+        val token = task.result.find { it.fpanLastFour == last4Digits }
+        token?.let {
+          Log.i("getCardStatus", "Card Token State: ${it.tokenState}")
+          promise.resolve(
+            getCardStatusCode(token.tokenState)
+          )
+        } ?:  promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
+      }
+      .addOnFailureListener { e -> promise.reject("getCardStatus function failed", e) }
+      .addOnCanceledListener {
+        promise.reject(
+          "Reject",
+          "Card status retrieval canceled"
+        )
+      }
+  }
+
+  @ReactMethod
+  override fun getCardTokenStatus(tsp: String, tokenRefId: String, promise: Promise) {
+    if (!ensureTapAndPayClientInitialized()) {
+      promise.reject("Initialization error", "TapAndPay client initialization failed")
+      return
+    }
+
+    tapAndPayClient!!.getTokenStatus(getTokenServiceProvider(tsp), tokenRefId)
+      .addOnCompleteListener { task ->
+        if (!task.isSuccessful || task.result == null) {
+          promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
+          return@addOnCompleteListener
+        }
+        val token = task.result
+        token?.let {
+          promise.resolve(
+            getCardStatusCode(token.tokenState)
+          )
+        } ?:  promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
+      }
+      .addOnFailureListener { e -> promise.reject("getCardStatus function failed", e) }
+      .addOnCanceledListener {
+        promise.reject(
+          "Reject",
+          "Card status retrieval canceled"
+        )
+      }
+  }
+
+
+  @ReactMethod
+  override fun addCardToWallet(
+    data: ReadableMap, promise: Promise
+  ) {
+    if (!ensureTapAndPayClientInitialized()) return
+
+    try {
+      val cardData = data.toCardData() ?: return promise.reject("Reject: ", "Insufficient data")
+
+      val cardNetwork = getCardNetwork(cardData.network)
+      val tokenServiceProvider = getTokenServiceProvider(cardData.network)
+      if (cardNetwork == 1000 || tokenServiceProvider == 1000) {
+        return promise.reject("Reject: ", "Invalid card network")
+      }
+
+      val pushTokenizeRequest = PushTokenizeRequest.Builder()
+        .setOpaquePaymentCard(cardData.opaquePaymentCard.toByteArray(Charset.forName("UTF-8")))
+        .setNetwork(cardNetwork)
+        .setTokenServiceProvider(tokenServiceProvider)
+        .setDisplayName(cardData.cardHolderName)
+        .setLastDigits(cardData.lastDigits)
+        .setUserAddress(cardData.userAddress)
+        .build()
+
+      tapAndPayClient?.pushTokenize(
```

Comment:
```suggestion
      tapAndPayClient!!.pushTokenize(
```

I feel like non-null asserts would be better here. If `tapAndPayClient` change to null between here and the check at the beginning of the function, it would leave a trace and you'd know that some assumption was wrong. Now it will fail silently.